ten_quick_tips_live_coding.md

```diff
@@ -53,46 +53,46 @@ university courses involving the teaching of programming or related techniques e
-(https://doi.org/10.1145/3279720.3279725 and references therein, as well as https://dl.acm.org/doi/10.1145/3373165.3373182).
-So far, the technique show that live-coding is as good as if not better than static code examples.
+demonstrating the programming process and contrasting this with the use of static code on slides
+(see https://doi.org/10.1145/3279720.3279725 and references therein, as well as https://dl.acm.org/doi/10.1145/3373165.3373182).
+So far, the technique show that live-coding is as good as if not better than using static code examples.
 
 The ten quick tips described below are aimed at those interested in applying the technique in to their own teaching.
-These complement the "Ten quick tips for delivering programming lessons"
+They are meant to complement the "Ten quick tips for delivering programming lessons"
 https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007433.
 
+
 ## Tip 1. Go slowly.
 
 Say out loud what you are doing while you do it
 for every command you type,
 every word of code you write,
-and every menu item or website button you click,
-and then point to the command and its output on the screen and optionally go through it a second time.
+and every menu item or website button you click.
+Then, point to the command and its output on the screen
+and optionally go through it a second time.
 This not only slows you down,
 it allows learners who are following along to copy what you do,
 or to catch up, even when they are looking at their screen while doing it.
 If the output of your command or code makes what you just typed disappear from view,
 scroll back up so learners can see it again.
 
-*Do not copy-paste code from your lesson script.*
-[FIXME: explain why not.]
+Do not copy-paste code from your lesson material and/or ask your learners to do the same. It will make you go fast with the risk of you not explaining the thought process behind the code.
 
 ## Tip 2. Mirror your learner's environment.
 
-Try to create an environment that is as similar as possible to what your learners have to reduce cognitive load.
-[FIXME: explain or include reference for "cognitive load".]
+Try to create an environment that is as similar as possible to what your learners have. If learners have to work in a different environment that you adds a mental effort that does not contribute to learning. Cognitive psychological theory calls this extraneous cognitive load (https://doi.org/10.1007/s10648-019-09465-5).
 You may have personalised your environment with a very simple or rather fancy Unix prompt,
 colour schemes for your development environment, keyboard shortcuts, etc.
+
 Your learners usually won't have all of this,
 so try to create an environment that mirrors what your learners have.
 
-Similarly, avoid using keyboard shortcuts.
-[FIXME: a sentence of explanation.]
+Similarly, avoid using keyboard shortcuts as these will hide the action(s) you are performing and learners may not know about them.
```

Comment:
If you _must_ use them, use a tool like KeyCastr (https://github.com/keycastr/keycastr) or your computer's accessibility tools to echo keystrokes to the screen.

Reply:
keycastr is mac only and I hesitate to suggest specific applications. So adjusted to "If you must use them, use a keystroke visualizer tool or your computer's accessibility tools to echo keystrokes to the screen."